src/SampleWorker/SampleJobProcessor.cs

```diff
@@ -27,7 +27,7 @@ public Task<long> GetTargetQueueLength()
             return Task.FromResult(0L);
         }
 
-        public void Initialize()
+        public void Finalize()
```

Comment:
The "Finalize" name can be mis-iterpreted and confused with "Finalizers" in C#.
Plus, now that we're creating such methods, we can add other methods too, such as initialization support, or notification for other events.

I'm proposing "OnJobCompleted" and "OnJobStarted" method names, not sure if they're good. Or, maybe a more generic name such as "OnJobStateChanged" with a parameter.

Reply:
Maybe "ProcessJobInitialization" and "ProcessJobCleanup" can be better proposals, they conform better with method naming (being verbs)